zerver/tests/test_import_export.py

```diff
@@ -594,48 +594,60 @@ def test_zulip_realm(self) -> None:
+        pm_c_msg_id = self.send_personal_message(hamlet, othello)
+        pm_d_msg_id = self.send_personal_message(iago, hamlet)
 
-        data = read_json("realm.json")
+        self.export_realm_and_create_auditlog(
+            realm,
+            export_type=RealmExport.EXPORT_FULL_WITH_CONSENT,
+            exportable_user_ids=exportable_user_ids,
+        )
 
-        exported_user_emails = self.get_set(data["zerver_userprofile"], "delivery_email")
-        self.assertIn(self.example_email("iago"), exported_user_emails)
-        self.assertIn(self.example_email("hamlet"), exported_user_emails)
+        realm_data = read_json("realm.json")
+
+        exported_user_emails = self.get_set(realm_data["zerver_userprofile"], "delivery_email")
+        self.assertIn(cordelia.delivery_email, exported_user_emails)
+        self.assertIn(hamlet.delivery_email, exported_user_emails)
         self.assertNotIn("default-bot@zulip.com", exported_user_emails)
-        self.assertNotIn(self.example_email("cordelia"), exported_user_emails)
+        self.assertNotIn(iago.delivery_email, exported_user_emails)
+        self.assertNotIn(polonius.delivery_email, exported_user_emails)
 
-        dummy_user_emails = self.get_set(data["zerver_userprofile_mirrordummy"], "delivery_email")
-        self.assertIn(self.example_email("cordelia"), dummy_user_emails)
-        self.assertIn(self.example_email("othello"), dummy_user_emails)
+        dummy_user_emails = self.get_set(
+            realm_data["zerver_userprofile_mirrordummy"], "delivery_email"
+        )
+        self.assertIn(iago.delivery_email, dummy_user_emails)
+        self.assertIn(othello.delivery_email, dummy_user_emails)
+        self.assertIn(polonius.delivery_email, dummy_user_emails)
         self.assertIn("default-bot@zulip.com", dummy_user_emails)
-        self.assertNotIn(self.example_email("iago"), dummy_user_emails)
-        self.assertNotIn(self.example_email("hamlet"), dummy_user_emails)
+        self.assertNotIn(cordelia.delivery_email, dummy_user_emails)
+        self.assertNotIn(hamlet.delivery_email, dummy_user_emails)
 
-        data = read_json("messages-000001.json")
+        message_data = read_json("messages-000001.json")
 
-        exported_message_ids = self.get_set(data["zerver_message"], "id")
+        exported_message_ids = self.get_set(message_data["zerver_message"], "id")
         self.assertNotIn(pm_a_msg_id, exported_message_ids)
         self.assertIn(pm_b_msg_id, exported_message_ids)
         self.assertIn(pm_c_msg_id, exported_message_ids)
         self.assertIn(pm_d_msg_id, exported_message_ids)
 
+        personal_recipient_type_ids = (
+            r["type_id"] for r in realm_data["zerver_recipient"] if r["type"] == Recipient.PERSONAL
+        )
+        for user_profile_id in [cordelia.id, hamlet.id, iago.id, othello.id, polonius.id]:
+            self.assertIn(user_profile_id, personal_recipient_type_ids)
```

Comment:
the actual important change in this test - check that `user_profile.recipient` is not broken for users that weren't in `exportable_user_ids`.